peps/pep-0750.rst

```diff
@@ -0,0 +1,894 @@
+PEP: 750
+Title: Tag Strings For Writing Domain-Specific Languages
+Author: Jim Baker <jim.baker@python.org>, Guido van Rossum <guido@python.org>, Paul Everitt <pauleveritt@me.com>
+Sponsor: Lysandros Nikolaou <lisandrosnik@gmail.com>
+Status: Draft
+Type: Standards Track
+Created: 08-Jul-2024
+Python-Version: 3.14
+
+Abstract
+========
+
+This PEP introduces tag strings for custom, repeatable string processing. Tag strings
+are an extension to f-strings, with a custom function -- the "tag" -- in place of the
+``f`` prefix. This function can then provide rich features such as safety checks, lazy
+evaluation, domain-specific languages (DSLs) for web templating, and more.
+
+Tag strings are similar to `JavaScript tagged template literals <https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals#tagged_templates>`_
+and related ideas in other languages. The following tag string usage shows how similar it is to an ``f`` string, albeit
+with the ability to process the literal string and embedded values:
+
+.. code-block:: python
+
+    name = "World"
+    greeting = greet"hello {name}"
+    assert greeting == "Hello WORLD!"
```

Comment:
Shouldn't the case on `World` match what is contained in `main`?

Reply:
OK. What's confusing is that the greet tag is defined below this example.

Reply:
It was suggested during review to lead the narrative with a usage, though all the others are definition->usage. Want me to put definition first?

Reply:
I think it is fine @pauleveritt to leave in this order but add a brief docstring to the function that is right after.